analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/ProjectSerializer.kt

```diff
@@ -42,10 +45,10 @@ object ProjectSerializer {
      * @param compress whether the output should be GZIP compressed
      */
     @Throws(IOException::class)
-    fun serializeProject(project: Project, out: OutputStream, compress: Boolean) {
-        val wrappedOut = if (compress) GZIPOutputStream(out) else out
+    fun serializeProject(project: Project, out: OutputStream, compress: Boolean, isOutputFileSpecified: Boolean = false) {
+        val wrappedOut = if (compress && isOutputFileSpecified) GZIPOutputStream(out) else out
```

Comment:
What about setting compress to false somewhere at a former step in case the output file is not specified?

I would also either log a warning in case no output file is specified but compress is set or I would just accept the output to be compressed even as stream to the terminal / not having an output file specified. Was there a specific reason that this is not possible right now?

Reply:
Applying compression to json data that gets written to stdout would require us to rewrite the logic for handling GZIP-Inputstreams. We agreed to take this as a feature request #3444, but will focus on other tickets in the meantime.